File: EmptyProject/WorkflowsTrainingTwo/WorkflowsTraining/Parsing/Statements/Print/PrintNode.cs
using WorkflowsTraining.Parsing.Expressions.Interfaces;
using WorkflowsTraining.Parsing.Statements.Interfaces;

namespace WorkflowsTraining.Parsing.Statements.Print;

public class PrintNode(IExpressionNode value) : IStatementNode
{
    private IExpressionNode Value { get; } = value;

    public void Execute()
    {
        Console.WriteLine(Value.Resolve());
    }
}

File: EmptyProject/WorkflowsTrainingTwo/WorkflowsTrainingCPlus/Parsing/Statements/Print/PrintNode.cpp
#include "PrintNode.h"
#include <iostream>

namespace WorkflowsTraining::Parsing::Statements::Print {
    PrintNode::PrintNode(std::unique_ptr<Expressions::Interfaces::IExpressionNode> value)
        : value_(std::move(value)) {
    }

    void PrintNode::Execute() {
        std::cout << "Executing PrintNode: ";

        auto resolvedValue = value_->Resolve();

        std::visit([](auto&& arg) {
            using T = std::decay_t<decltype(arg)>;

            if constexpr (std::is_same_v<T, bool>) {
                std::cout << (arg ? "true" : "false");
            }
            else if constexpr (std::is_same_v<T, int>) {
                std::cout << arg;
            }
            else if constexpr (std::is_same_v<T, double>) {
                std::cout << arg;
            }
            else if constexpr (std::is_same_v<T, std::string>) {
                std::cout << arg;
            }
            else {
                std::cout << "[Unsupported type for print]";
            }
            }, resolvedValue);
        std::cout << std::endl;
    }
}